File: School42/playground/-/languages/Prolog/celsius2fahrenheit.pl
convert_temperatures(Fahr) :-
    Fahr =< 300,
    Celsius is (5.0 / 9.0) * (Fahr - 32),
    format('°F: ~d  °C: ~2f~n', [Fahr, Celsius]),
    NextFahr is Fahr + 20,
    convert_temperatures(NextFahr).
convert_temperatures(_).

File: School42/playground/-/languages/Haskell/celsius2fahrenheit.hs
main :: IO ()
main = loop 0
  where
    loop :: Int -> IO ()
    loop fahr
      | fahr <= 300 = do
          let celsius = (5.0 / 9.0) * fromIntegral (fahr - 32)
          putStrLn $ "°F: " ++ show fahr ++ "  °C: " ++ show celsius
          loop (fahr + 20)
      | otherwise = return ()
